### DISCUSSION TITLE
Please consider adding a REST API support

### DISCUSSION THREAD

**Discussion by Diniboy1123** [librespot-org/librespot#794] [Feature Requests]
Hi,

Similarly to librespot-java, I believe it would be amazing if this project could expose a small internal webserver with basic functions such as play a specific song, toggle playback, change volume, seeking and shuffling. Not to mention it would be great if there would be a websocket too sending track paused/resumed/stopped etc events.

I assume this has been requested before, but I couldn't find any related feature requests yet. ☺️

**Comment by ashthespy** [comment]
Have a look at https://github.com/librespot-org/librespot/issues/457 for some thoughts :-)

**Reply by Diniboy1123** [reply]
I see, thanks. So the only option right now is to use an external lib. I assume this is to reduce the package size. But if I saw it correctly you already have quite a few web related deps, so one could create an API from within the project.

What's the reason why it was never added to librespot?

**Reply by ashthespy** [reply]
> What's the reason why it was never added to librespot?

This goes back to `librespot` the library vs `librespot` the binary.. 
The binary that is provided is a more as an example of a "minimal" Connect implementation.  It wasn't meant to be directly controlled. But, the library provides things you need to wrap it into a client that can expose an API as desired.
